docs/_src/tutorials/tutorials/16.md

```diff
@@ -163,6 +191,61 @@ prediction = pipe.run(
 print_answers(prediction, details="high")
 ```
 
+## Wrapping it up in an indexing pipeline
+
+
+```python
+from pathlib import Path
+from haystack.pipelines import Pipeline
+from haystack.nodes import TextConverter, PreProcessor, FileTypeClassifier, PDFToTextConverter, DocxToTextConverter
+```
+
+
+```python
+# we need to set the batch_size so each batch fits into memory
+index_time_document_classifier = TransformersDocumentClassifier(
+    model_name_or_path="cross-encoder/nli-distilroberta-base",
+    task="zero-shot-classification",
+    labels=["music", "natural language processing", "history"], 
+    batch_size=16
+)
+```
+
+
+```python
+file_type_classifier = FileTypeClassifier()
+text_converter = TextConverter()
+pdf_converter = PDFToTextConverter()
+docx_converter = DocxToTextConverter()
+
+indexing_pipeline_with_classification = Pipeline()
+indexing_pipeline_with_classification.add_node(component=file_type_classifier, name="FileTypeClassifier", inputs=["File"])
+indexing_pipeline_with_classification.add_node(component=text_converter, name="TextConverter", inputs=["FileTypeClassifier.output_1"])
+indexing_pipeline_with_classification.add_node(component=pdf_converter, name="PdfConverter", inputs=["FileTypeClassifier.output_2"])
+indexing_pipeline_with_classification.add_node(component=docx_converter, name="DocxConverter", inputs=["FileTypeClassifier.output_4"])
```

Comment:
As the tutorial is not about the file converters having three of them seems a bit distracting. However, I see that this is needed because of the dataset that you use here. Not sure about the exact files in there but how about only selecting one format (e.g. txt) **before** running the indexing pipeline and getting rid of those extra converters? Makes only sense if there are enough files in one of the formats.

Reply:
Yes, that's basically the problem. I took the preprocessing data, which consists of one pdf regarding nlp, one docx regarding music and one txt regarding history. I'm not sure if converting two of them and creating a nother dataset is worth it. But I totally agree that this part looks not to snappy.

Reply:
Ok let's stay pragmatic here and keep it for now as it is